### PR TITLE
*: remove non-prepared plan cache

### PR DESCRIPTION
**Pull request by coocood**
## What have you changed? (mandatory)

Remove non-prepared plan cache。

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

non-prepared plan cache is not useable if we can only do full string match.
And further development doesn't worth the effort.

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

Yes

## Does this PR affect tidb-ansible update? (mandatory)

Yes

## Does this PR need to be added to the release notes? (mandatory)

release note:
Remove `plan-cache` config option.

### CODE REVIEW

**Comment by coocood** [comment]
@zz-jason @tiancaiamao PTAL

File: session/session.go [hunk 1]
```diff
@@ -767,86 +767,45 @@ func (s *session) execute(ctx context.Context, sql string) (recordSets []ast.Rec
...
-			// Step3: Cache the physical plan if possible.
-			if planCacheEnabled && stmt.Cacheable && len(stmtNodes) == 1 && !s.GetSessionVars().StmtCtx.HistogramsNotLoad() {
-				plan.GlobalPlanCache.Put(cacheKey, plan.NewSQLCacheValue(stmtNode, stmt.Plan, stmt.Expensive))
-			}
-
-			// Step4: Execute the physical plan.
-			if recordSets, err = s.executeStatement(ctx, connID, stmtNode, stmt, recordSets); err != nil {
-				return nil, errors.Trace(err)
-			}
+		// Step4: Execute the physical plan.
```
Comment (zz-jason) [inline session/session.go]: this is "step3" now.

File: util/kvcache/sharded_lru.go [hunk 2]
```diff
@@ -1,58 +0,0 @@
-// Copyright 2017 PingCAP, Inc.
```
Comment (zz-jason) [inline util/kvcache/sharded_lru.go]: this could be useful, maybe we can save it.

**Comment by zz-jason** [comment]
LGTM

**Comment by coocood** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Comment by zz-jason** [comment]
@coocood Please add the "release-note" label
cc @shenli

**Comment by tiancaiamao** [comment]
And update the document too.

**Comment by tiancaiamao** [comment]
And update the document too.

**Comment by tiancaiamao** [comment]
And update the document too.

**Pull request merged**